hbase-server/src/main/java/org/apache/hadoop/hbase/master/http/MasterHbckServlet.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.client.RegionInfo;
+import org.apache.hadoop.hbase.master.HMaster;
+import org.apache.hadoop.hbase.master.HbckChore;
+import org.apache.hadoop.hbase.master.http.gson.GsonFactory;
+import org.apache.hadoop.hbase.master.janitor.CatalogJanitor;
+import org.apache.hadoop.hbase.master.janitor.Report;
+import org.apache.hadoop.hbase.util.Pair;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hbase.thirdparty.com.google.gson.Gson;
+
+@InterfaceAudience.Private
+public class MasterHbckServlet extends HttpServlet {
+
+  public static final String START_TIMESTAMP = "start_timestamp";
+  public static final String END_TIMESTAMP = "end_timestamp";
+  public static final String INCONSISTENT_REGIONS = "inconsistent_regions";
+  public static final String ORPHAN_REGIONS_ON_RS = "orphan_regions_on_rs";
+  public static final String ORPHAN_REGIONS_ON_FS = "orphan_regions_on_fs";
+  public static final String HOLES = "holes";
+  public static final String OVERLAPS = "overlaps";
+  public static final String UNKNOWN_SERVERS = "unknown_servers";
+  public static final String EMPTY_REGIONINFO = "empty_regioninfo";
+
+  private static final long serialVersionUID = 1L;
+  private static final Logger LOG = LoggerFactory.getLogger(MasterHbckServlet.class);
+  private static final Gson GSON = GsonFactory.buildGson();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response)
+    throws ServletException, IOException {
+    final HMaster master = (HMaster) getServletContext().getAttribute(HMaster.MASTER);
+    if (!master.isInitialized()) {
+      LOG.warn("Master is not initialized yet");
+      sendError(response, HttpServletResponse.SC_SERVICE_UNAVAILABLE,
+        "master is not initialized yet");
+      return;
+    }
+    final HbckChore hbckChore = master.getHbckChore();
+    if (hbckChore == null || hbckChore.isDisabled()) {
+      LOG.warn("Hbck chore is disabled");
+      sendError(response, HttpServletResponse.SC_SERVICE_UNAVAILABLE, "Hbck chore is disabled");
+      return;
+    }
+    if (!Boolean.parseBoolean(request.getParameter("cache"))) {
+      try {
+        master.getMasterRpcServices().runHbckChore(null, null);
+      } catch (org.apache.hbase.thirdparty.com.google.protobuf.ServiceException se) {
+        LOG.warn("Failed generating a new hbck chore report; using cache", se);
+      }
+      try {
+        master.getMasterRpcServices().runCatalogScan(null, null);
+      } catch (org.apache.hbase.thirdparty.com.google.protobuf.ServiceException se) {
+        LOG.warn("Failed generating a new catalogjanitor report; using cache", se);
+      }
+    }
+    Map<String, Object> result = new HashMap<>();
+    result.put(START_TIMESTAMP, hbckChore.getCheckingStartTimestamp());
+    result.put(END_TIMESTAMP, hbckChore.getCheckingEndTimestamp());
```

Comment:
Our HBCK chore should expose a report so that a reader can get a static view... or a read-write lock so that a reader can block a writer from clobbering its view.